src/builders/browser/index.ts

```diff
@@ -17,42 +18,18 @@ export class SingleSpaBrowserBuilder extends BrowserBuilder {
     projectRoot: Path,
     host: virtualFs.Host<fs.Stats>,
     options: SingleSpaBuilderSchema,
-  ): Configuration {
-
-    const libraryName = options.libraryName || this.context.targetSpecifier && this.context.targetSpecifier.project;
-
+  ): Configuration {    
     // Disable es2015 polyfills
     // tslint:disable-next-line: max-line-length
     // https://github.com/angular/angular-cli/blob/3d8064bb64d72557474a7484f1b85eaf35788916/packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts#L56
     options.es5BrowserSupport = false;
-
     // Generate Angular CLI's default Browser Webpack cofiguration
-    const config = super.buildWebpackConfig(root, projectRoot, host, options);
-
-    // Remove bundles
-    delete config.entry.polyfills;
-    delete config.entry.styles;
-    delete config.optimization.runtimeChunk;
-    delete config.optimization.splitChunks;
-
-    // Remove Angular's IndexHtmlWebpackPlugin from build
-    const indexHtmlWebpackPluginIndex = config.plugins.findIndex(
-      plugin => (plugin as any).constructor.name === 'IndexHtmlWebpackPlugin',
-    );
-
-    if (indexHtmlWebpackPluginIndex > -1) {
-      config.plugins.splice(indexHtmlWebpackPluginIndex, 1);
-    }
+    const config = super.buildWebpackConfig(root, projectRoot, host, options);    
 
-    return webpackMerge.smart(config, {
-      output: {
-        library: libraryName,
-        libraryTarget: options.libraryTarget as any,
-      },
-      externals: {
-        'zone.js': 'Zone',
-      },
-    });
+    // Delegate the building of the webpack config to the new builder.
+    // Builder based on custom builders implmented by @meltedspark
+    // https://github.com/meltedspark/angular-builders
+    return SingleSpaWebpackBuilder.buildWebpackConfig(root, options.singleSpaWebpackConfig, config, options, this.context);
```

Comment:
Being honest, I think that this pull request has made the code significantly more complicated than it needs to be. Here is what we could have done:

```js
const extraConfig = options.singleSpaWebpackConfig ? require(options.singleSpaWebpackConfig) : {}
return webpackMerge.smart(config, {
  output: {
    library: libraryName,
    libraryTarget: options.libraryTarget as any,
  },
  externals: {
    'zone.js': 'Zone',
  },
}, extraConfig);
```

Is there a reason why this implementation ^ wouldn't work?

Reply:
I don't see why this wouldn't work if you don't want the additional flexibility. I followed the meltedspark implementation and refactored the actual merge into helper classes so that it could be re-used in other builders should we find the need to implement them.

As it stands the dev server builder is dependent on the browser build, which IMO is a bit of a smell. I should, as part of this PR, break that dependency and make the dev-server buider use the merger.

Reply:
Just added a commit that makes the dev server builder use the helper class rather than the browser builder